gql/client.py

```diff
@@ -880,6 +937,112 @@ def execute(
+                    if req.variable_values is not None
+                    else req
+                    for req in reqs
+                ]
+
+        results = self.transport.execute_batch(reqs, **kwargs)
+
+        # Unserialize the result if requested
+        if self.client.schema:
+            if parse_result or (parse_result is None and self.client.parse_results):
+                for result in results:
+                    result.data = parse_result_fn(
+                        self.client.schema,
+                        req.document,
+                        result.data,
+                        operation_name=req.operation_name,
+                    )
+
+        return results
+
+    def execute_batch(
+        self,
+        reqs: List[GraphQLRequest],
+        serialize_variables: Optional[bool] = None,
+        parse_result: Optional[bool] = None,
+        get_execution_result: bool = False,
+        **kwargs,
+    ) -> Union[List[Dict[str, Any]], List[ExecutionResult]]:
+        """Execute the provided document AST synchronously using
+        the sync transport.
+
+        Raises a TransportQueryError if an error has been returned in
+            the ExecutionResult.
+
+        :param document: GraphQL query as AST Node object.
+        :param variable_values: Dictionary of input parameters.
+        :param operation_name: Name of the operation that shall be executed.
+        :param serialize_variables: whether the variable values should be
+            serialized. Used for custom scalars and/or enums.
+            By default use the serialize_variables argument of the client.
+        :param parse_result: Whether gql will unserialize the result.
+            By default use the parse_results argument of the client.
+        :param get_execution_result: return the full ExecutionResult instance instead of
+            only the "data" field. Necessary if you want to get the "extensions" field.
+
+        The extra arguments are passed to the transport execute method."""
+
+        # Validate and execute on the transport
+        results = self._execute_batch(
+            reqs,
+            serialize_variables=serialize_variables,
+            parse_result=parse_result,
+            **kwargs,
+        )
+
+        for result in results:
+            # Raise an error if an error is returned in the ExecutionResult object
+            if result.errors:
+                raise TransportQueryError(
+                    str_first_element(result.errors),
+                    errors=result.errors,
+                    data=result.data,
+                    extensions=result.extensions,
+                )
+
+            assert (
+                result.data is not None
+            ), "Transport returned an ExecutionResult without data or errors"
+
+        if get_execution_result:
+            return results
+
+        return [result.data for result in results]  # type: ignore
```

Comment:
Please remove the `# type: ignore` here.
I realize that the problem is that mypy is not smart enough to realize that the answer is of type `List[Dict[str, Any]]` instead of `List[Optional[Dict[str, Any]]]` even with the asserts above.
This should be solved by using the `cast` method to cast the answer to `List[Dict[str, Any]]`

Reply:
Agree, but anyways I don't really like the type system of python. it's horrible, I tried to not worry too much about it because it simply can't handle any simple case. But no problem about the casting.

Reply:
I know, but it helps sometimes. I prefer to avoid using ignore as much as possible if there is a clean way to do it, it avoids the mental load when you read the code months later and you don't know if it's legit to ignore the type here.